src/main/java/jp/classmethod/aws/gradle/AwsPluginExtension.java

```diff
@@ -70,29 +74,37 @@
 	private AWSCredentialsProvider credentialsProvider;
 	
 	
-	public AWSCredentialsProvider newCredentialsProvider(String profileName) {
-		if (credentialsProvider != null) {
-			return credentialsProvider;
-		}
+	public AWSCredentialsProvider newCredentialsProvider(String profileName, String roleArn) {
+		return credentialsProvider != null ? credentialsProvider : buildCredentialsProvider(profileName, roleArn);
+	}
+	
+	private AWSCredentialsProvider buildCredentialsProvider(String profileName, String roleArn) {
+		List<AWSCredentialsProvider> providers = new ArrayList<>();
+		providers.add(new EnvironmentVariableCredentialsProvider());
+		providers.add(new SystemPropertiesCredentialsProvider());
+		
 		String profileNameToUse = profileName != null ? profileName : this.profileName;
-		if (Strings.isNullOrEmpty(profileNameToUse) == false) {
-			List<AWSCredentialsProvider> providers = new ArrayList<AWSCredentialsProvider>();
-			providers.add(new EnvironmentVariableCredentialsProvider());
-			providers.add(new SystemPropertiesCredentialsProvider());
+		if (!Strings.isNullOrEmpty(profileNameToUse)) {
```

Comment:
this change has introduced a bug: now ProfileCredentialsProvider is only added if profileName is explicitly specified. In my setup I use profiles, but I don't want to specify it in build.gradle, instead I have AWS_PROFILE env variable set. This used to work and no longer does, so I had to downgrade to 0.39 for now